session/txn.go

```diff
@@ -404,11 +368,13 @@ func (s *session) getTxnFuture(ctx context.Context) *txnFuture {
 // HasDirtyContent checks whether there's dirty update on the given table.
 // Put this function here is to avoid cycle import.
 func (s *session) HasDirtyContent(tid int64) bool {
-	x := s.GetSessionVars().TxnCtx.DirtyDB
-	if x == nil {
+	if s.txn.Transaction == nil {
 		return false
 	}
-	return !x.(*executor.DirtyDB).GetDirtyTable(tid).IsEmpty()
+	seekKey := tablecodec.EncodeTablePrefix(tid)
+	it, err := s.txn.GetMemBuffer().Iter(seekKey, nil)
+	_ = err
```

Comment:
print error log instead of ignoring it?

Reply:
MemBuffer.Iter will never return an error.